### PR TITLE
Reduce bundle size by minimizing use of `react-icons`

### PR DESCRIPTION
**Pull request by dan-mba**
## Changes proposed
Using icons from one of `react-icons` icon sets causes the js file for the whole set to be bundled.
This PR replaces these icons in many places with ones from `heroicons` which was already used elsewhere in the project.
For the brand icons used in the footer, I extracted them from from `react-icons/fa`. 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots
Current bundle size:

```
Route (pages)                                 Size     First Load JS
┌ ● /                                         7.32 kB         547 kB
├   /_app                                     0 B             538 kB
├ λ /[username]                               907 B           587 kB
├ ○ /404                                      1.17 kB         539 kB
├ λ /account/no-profile                       1.2 kB          539 kB
├ λ /account/statistics                       110 kB          650 kB
├ λ /api/account/statistics                   0 B             538 kB
├ λ /api/auth/[...nextauth]                   0 B             538 kB
├ λ /api/auth/db-adapter                      0 B             538 kB
├ λ /api/discover/tags                        0 B             538 kB
├ λ /api/events                               0 B             538 kB
├ λ /api/revalidate                           0 B             538 kB
├ λ /api/statistics/today                     0 B             538 kB
├ λ /api/statistics/totals                    0 B             538 kB
├ λ /api/system/reload                        0 B             538 kB
├ λ /api/users                                0 B             538 kB
├ λ /api/users/[username]                     0 B             538 kB
├ λ /api/users/[username]/links/[url]         0 B             538 kB
├ λ /auth/signin                              1.29 kB         540 kB
├ ○ /changelog                                2.18 kB         540 kB
├ ○ /docs                                     3.32 kB         542 kB
├ ○ /docs/advanced/single-user-mode           2.03 kB         541 kB
├ ○ /docs/community-resources                 2.69 kB         542 kB
├ ○ /docs/contributing/automated-tests        2.17 kB         541 kB
├ ○ /docs/contributing/commits                2.15 kB         541 kB
├ ○ /docs/contributing/hacktoberfest          2.34 kB         542 kB
├ ○ /docs/contributing/labels                 2.74 kB         542 kB
├ ○ /docs/contributing/reviewers              3.59 kB         543 kB
├ ○ /docs/contributing/storybook              2.18 kB         541 kB
├ ○ /docs/docs-style-guide                    2.36 kB         542 kB
├ ○ /docs/environments/environment-variables  3.61 kB         543 kB
├ ○ /docs/environments/github-ui              1.68 kB         541 kB
├ ○ /docs/environments/gitpod                 3.25 kB         542 kB
├ ○ /docs/environments/linkfree-cli           2.83 kB         542 kB
├ ○ /docs/environments/local-development      3.07 kB         542 kB
├ ○ /docs/environments/local-with-atlas       2.87 kB         542 kB
├ ○ /docs/faqs                                2.04 kB         541 kB
├ ○ /docs/full-profile-example                3.17 kB         542 kB
├ ○ /docs/how-to-guides/bio                   3.09 kB         542 kB
├ ○ /docs/how-to-guides/editing               3.81 kB         543 kB
├ ○ /docs/how-to-guides/events                4.24 kB         543 kB
├ ○ /docs/how-to-guides/links                 3.69 kB         543 kB
├ ○ /docs/how-to-guides/milestones            3.6 kB          543 kB
├ ○ /docs/how-to-guides/socials-shortcuts     3.33 kB         543 kB
├ ○ /docs/how-to-guides/statistics            2.46 kB         542 kB
├ ○ /docs/how-to-guides/tags                  3.15 kB         542 kB
├ ○ /docs/how-to-guides/testimonials          3.48 kB         543 kB
├ ○ /docs/maintainers                         3.89 kB         543 kB
├ ○ /docs/map                                 2.74 kB         542 kB
├ ○ /docs/profile-tips                        2.4 kB          542 kB
├ ○ /docs/quickstart                          5.65 kB         545 kB
├ λ /events                                   3.61 kB         577 kB
├ ○ /icons                                    1.64 MB        2.18 MB
├ ○ /maintainers                              1.67 kB         540 kB
├ ● /map                                      2.27 kB         541 kB
├ ○ /playground                               15.9 kB         602 kB
├ ○ /premium                                  1.85 kB         542 kB
├ ○ /roadmap                                  2.44 kB         541 kB
├ ● /search                                   3.21 kB         576 kB
└ ○ /terms                                    1.19 kB         539 kB
+ First Load JS shared by all                 550 kB
  ├ chunks/framework-ce84985cd166733a.js      45.2 kB
  ├ chunks/main-6727a303e2c2ac9e.js           34.1 kB
  ├ chunks/pages/_app-61cf76f3f16b2adb.js     457 kB
  ├ chunks/webpack-b5fae74c2c0e5831.js        2.3 kB
  └ css/750be684b6baab38.css                  11.5 kB
```

Build from this PR:

```
Route (pages)                                 Size     First Load JS
┌ ● /                                         6.96 kB         546 kB
├   /_app                                     0 B             107 kB
├ λ /[username]                               913 B           588 kB
├ ○ /404                                      1.17 kB         108 kB
├ λ /account/no-profile                       1.2 kB          108 kB
├ λ /account/statistics                       111 kB          220 kB
├ λ /api/account/statistics                   0 B             107 kB
├ λ /api/auth/[...nextauth]                   0 B             107 kB
├ λ /api/auth/db-adapter                      0 B             107 kB
├ λ /api/discover/tags                        0 B             107 kB
├ λ /api/events                               0 B             107 kB
├ λ /api/revalidate                           0 B             107 kB
├ λ /api/statistics/today                     0 B             107 kB
├ λ /api/statistics/totals                    0 B             107 kB
├ λ /api/system/reload                        0 B             107 kB
├ λ /api/users                                0 B             107 kB
├ λ /api/users/[username]                     0 B             107 kB
├ λ /api/users/[username]/links/[url]         0 B             107 kB
├ λ /auth/signin                              1.92 kB         109 kB
├ ○ /changelog                                2.18 kB         109 kB
├ ○ /docs                                     3.32 kB         110 kB
├ ○ /docs/advanced/single-user-mode           2.52 kB         110 kB
├ ○ /docs/community-resources                 3.17 kB         110 kB
├ ○ /docs/contributing/automated-tests        2.66 kB         110 kB
├ ○ /docs/contributing/commits                2.64 kB         110 kB
├ ○ /docs/contributing/hacktoberfest          2.83 kB         110 kB
├ ○ /docs/contributing/labels                 3.23 kB         110 kB
├ ○ /docs/contributing/reviewers              4.08 kB         111 kB
├ ○ /docs/contributing/storybook              2.66 kB         110 kB
├ ○ /docs/docs-style-guide                    2.85 kB         110 kB
├ ○ /docs/environments/environment-variables  1.84 kB         112 kB
├ ○ /docs/environments/github-ui              2.16 kB         109 kB
├ ○ /docs/environments/gitpod                 3.75 kB         111 kB
├ ○ /docs/environments/linkfree-cli           3.31 kB         110 kB
├ ○ /docs/environments/local-development      3.56 kB         111 kB
├ ○ /docs/environments/local-with-atlas       3.35 kB         110 kB
├ ○ /docs/faqs                                2.52 kB         110 kB
├ ○ /docs/full-profile-example                3.71 kB         111 kB
├ ○ /docs/how-to-guides/bio                   1.32 kB         111 kB
├ ○ /docs/how-to-guides/editing               4.3 kB          111 kB
├ ○ /docs/how-to-guides/events                2.47 kB         113 kB
├ ○ /docs/how-to-guides/links                 1.94 kB         112 kB
├ ○ /docs/how-to-guides/milestones            1.83 kB         112 kB
├ ○ /docs/how-to-guides/socials-shortcuts     1.57 kB         112 kB
├ ○ /docs/how-to-guides/statistics            660 B           111 kB
├ ○ /docs/how-to-guides/tags                  1.38 kB         111 kB
├ ○ /docs/how-to-guides/testimonials          1.73 kB         112 kB
├ ○ /docs/maintainers                         4.38 kB         111 kB
├ ○ /docs/map                                 3.23 kB         110 kB
├ ○ /docs/profile-tips                        2.88 kB         110 kB
├ ○ /docs/quickstart                          3.97 kB         114 kB
├ λ /events                                   4.19 kB         576 kB
├ ○ /icons                                    1.64 MB        2.18 MB
├ ○ /maintainers                              1.67 kB         109 kB
├ ● /map                                      3.87 kB         111 kB
├ ○ /playground                               15.4 kB         603 kB
├ ○ /premium                                  2.26 kB         109 kB
├ ○ /roadmap                                  2.49 kB         110 kB
├ ● /search                                   3.6 kB          143 kB
└ ○ /terms                                    1.19 kB         108 kB
+ First Load JS shared by all                 118 kB
  ├ chunks/framework-ce84985cd166733a.js      45.2 kB
  ├ chunks/main-3b80f74e2637dadb.js           33.9 kB
  ├ chunks/pages/_app-f96f384263b399e5.js     25.6 kB
  ├ chunks/webpack-54c03e940eb7a9b3.js        2.3 kB
  └ css/fbac15d0efbbb76c.css                  11.2 kB
```

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/7330"></a>

### CODE REVIEW

**Comment by eddiejaoude** [comment]
Oh wow this looks interesting 👍  the bundle size is much smaller - thank you! I will have a play/test

**Comment by dan-mba** [comment]
> Oh wow this looks interesting 👍 the bundle size is much smaller - thank you! I will have a play/test

The main thing to look at is the aesthetics.
This has zero changes on functionality and noticeable changes on looks.

File: pages/premium.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import Script from "next/script";
 import { useRouter } from "next/router";
 
-import { FaCheck } from "react-icons/fa";
+
```
Comment (eddiejaoude) [inline pages/premium.js:4]:
```suggestion
```

**Review by eddiejaoude — COMMENTED** [review]
Looks good! 

Q1. Do we still need the `react-icons`? I guess we do for the profiles

Q2. Do we need to include `@heroicons/react` just incase `react-icons` gets removed?

**Comment by dan-mba** [comment]
> Looks good!
> 
> Q1. Do we still need the `react-icons`? I guess we do for the profiles

Yes, FontAwesome is one of the few libraries that has icons for all the social media platforms.
And `react-icons` is the easiest way to access the full set of icons without doing a massive import.

> 
> Q2. Do we need to include `@heroicons/react` just in case `react-icons` gets removed?

Removing `react-icons` would be a major breaking change. I looked into it & there are no good other ways that I could find of allowing for the use of a large set of icons.

**Review by eddiejaoude — APPROVED** [review]
Let's GO!

**Pull request merged**